File: app/src/main/java/com/an/trailers/data/local/converter/VideoListTypeConverter.java
package com.an.trailers.data.local.converter;

import android.arch.persistence.room.TypeConverter;

import com.an.trailers.data.remote.model.Video;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class VideoListTypeConverter {

    @TypeConverter
    public List<String> fromString(String value) {
        Type listType = new TypeToken<List<String>>() {}.getType();
        List<String> videos = new Gson().fromJson(value, listType);
        return videos;
    }

    @TypeConverter
    public String fromList(List<String> videos) {
        return new Gson().toJson(videos);
    }

    public List<String> fromVideos(List<Video> videos) {
        if(videos == null) return Collections.EMPTY_LIST;
        List<String> videosList = new ArrayList<>();
        for(Video video : videos) {
            videosList.add(video.getKey());
        }
        return videosList;
    }
}

File: app/src/main/java/com/an/trailers/data/local/converter/VideoListTypeConverter.kt
package com.an.trailers.data.local.converter

import androidx.room.TypeConverter
import com.an.trailers.data.remote.model.Video
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class VideoListTypeConverter {
    @TypeConverter
    fun fromString(value: String): List<Video>? {
        val listType = object : TypeToken<List<Video>>() {}.type
        return Gson().fromJson<List<Video>>(value, listType)
    }

    @TypeConverter
    fun fromList(list: List<Video>?): String {
        val gson = Gson()
        return gson.toJson(list)
    }

    fun fromVideos(videos: List<Video>?): List<String> {
        if (videos == null) return emptyList()
        val videosList = ArrayList<String>()
        for (video in videos) {
            videosList.add(video.key)
        }
        return videosList
    }
}
